dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/changelog/ChangeLogServiceTest.java

```diff
@@ -0,0 +1,174 @@
+
+  private User importUser;
+
+  private TrackerImportParams importParams;
+
+  @Override
+  protected void initTest() throws IOException {
+    userService = _userService;
+    setUpMetadata("tracker/simple_metadata.json");
+    importUser = userService.getUser("M5zQapPyTZI");
+    importParams = TrackerImportParams.builder().userId(importUser.getUid()).build();
+    assertNoErrors(
+        trackerImportService.importTracker(
+            importParams, fromJson("tracker/event_and_enrollment.json")));
+  }
+
+  @Test
+  void shouldFailWhenEventDoesNotExist() {
+    assertThrows(
+        NotFoundException.class,
+        () -> changeLogService.getEventDataValueChangeLog(UID.of(CodeGenerator.generateUid())));
+  }
+
+  @Test
+  void shouldFailWhenEventOrgUnitIsNotAccessible() {
+    injectSecurityContextUser(manager.get(User.class, "o1HMTIzBGo7"));
+
+    assertThrows(
+        NotFoundException.class,
+        () -> changeLogService.getEventDataValueChangeLog(UID.of("D9PbzJY8bJM")));
+  }
+
+  @Test
+  void shouldFailWhenEventProgramIsNotAccessible() {
+    injectSecurityContextUser(manager.get(User.class, "o1HMTIzBGo7"));
+
+    assertThrows(
+        NotFoundException.class,
+        () -> changeLogService.getEventDataValueChangeLog(UID.of("G9PbzJY8bJG")));
+  }
+
+  @Test
+  void shouldFailWhenEventDataElementIsNotAccessible() {
+    /*    injectSecurityContextUser(manager.get(User.class, "FIgVWzUCkpw"));
+
+    assertThrows(
+        NotFoundException.class,
+        () -> changeLogService.getEventDataValueChangeLog(UID.of("pTzf9KYMk72")));*/
+    fail();
+  }
+
+  @Test
+  void shouldFailWhenProgramTrackedEntityTypeIsNotAccessible() {
+    fail();
+  }
+
+  @Test
+  void shouldReturnChangeLogListWhenDataValueIsCreated() {
+    fail();
+  }
+
+  @Test
+  void shouldReturnChangeLogListWhenDataValueIsDeleted() {
+    fail();
+  }
+
+  @Test
+  void shouldReturnChangeLogListWhenDataValueIsUpdated() throws NotFoundException, IOException {
+    injectSecurityContextUser(manager.get(User.class, "M5zQapPyTZI"));
+
+    Event event = manager.get(Event.class, "D9PbzJY8bJM");
+    assertNotNull(event);
+    String dataElementUid = event.getEventDataValues().iterator().next().getDataElement();
+    DataElement dataElement = manager.get(DataElement.class, dataElementUid);
+    assertNotNull(dataElement);
+
+    TrackerObjects trackerObjects = fromJson("tracker/event_and_enrollment.json");
+    trackerObjects.getEvents().stream()
+        .filter(e -> e.getEvent().equalsIgnoreCase(event.getUid()))
+        .findFirst()
+        .flatMap(
+            e ->
+                e.getDataValues().stream()
+                    .filter(
+                        dv -> dv.getDataElement().getIdentifier().equalsIgnoreCase(dataElementUid))
+                    .findFirst())
+        .ifPresent(dv -> dv.setValue("new value"));
+    assertNoErrors(trackerImportService.importTracker(importParams, trackerObjects));
+
+    List<EventDataValueChangeLog> changeLogList =
```

Comment:
```suggestion
    List<EventDataValueChangeLog> changeLogs =
```

the type is already in the type